src/core/reference/include/ngraph/runtime/reference/ctc_loss.hpp

```diff
@@ -105,9 +105,10 @@ void CTCLoss(const T* logits,
                         res = prevLogProb;
                     } else if (prevLogProb != -type_inf) {
                         if (res > prevLogProb)
-                            res = res + std::log1pf(std::exp(prevLogProb - res));
+                            res = res + static_cast<T>(std::log1pf(static_cast<float>(std::exp(prevLogProb - res))));
                         else
-                            res = prevLogProb + std::log1pf(std::exp(res - prevLogProb));
+                            res = prevLogProb +
+                                  static_cast<T>(std::log1pf(static_cast<float>(std::exp(res - prevLogProb))));
```

Comment:
or round?

Reply:
Cannot round() unless std::is_integral<T>::value is true.

Reply:
sure